content/07.architecture-and-representation.md

```diff
@@ -1 +1,22 @@
-## Tip 5: Choose an appropriate neural network architecture and data representation {#architecture}
+## Tip 5: Choose an appropriate data representation and neural network architecture {#architecture}
+Unfortunately, choosing how to represent your data and design your architecture is closer to an art than a science.
+While certain best practices have been established by the research community [@doi:10.1007/978-3-642-35289-8], architecture design choices remain largely problem-specific and are vastly empirical efforts requiring extensive experimentation.
+Furthermore, as deep learning is a quickly evolving field, many recommendations are often short-lived and frequently replaced by newer insights supported by recent empirical results.
+This is further complicated by the fact that many recommendations do not generalize well across different problems and datasets.
+With that being said, there are some general principles that are useful to follow when experimenting.
+
+First and foremost, use your knowledge of the available data and your question (see [Tip 4](#know-your-problem)) to inform your data representation and architectural design choices.
+For example, if your data is an array of measurements with no natural ordering of inputs (such as gene expression data), multilayer perceptrons (MLPs), which are the most basic type of neural network, may be effective.
+Similarly, if your data is comprised of images, convolutional neural networks (CNNs) are a good choice because they emphasize local structures and adjacency within the data.
+CNNs may also be a good choice for learning on sequences, as recent empirical evidence suggests they can outperform canonical sequence learning techniques such as recurrent neural networks (RNNs) and the closely related long short-term memory (LSTM) networks [arxiv:1803.01271]. 
+
+DL models can typically benefit from large amounts of labelled data to avoid overfitting (see [Tip 7](#overfitting)) and to achieve top performance on a task in hand.
+In the event that there is not enough data available to train your model, consider using transfer learning.
+In transfer learning, a model whose weights were generated by training on another dataset is used as the starting point for training [@tag:Yosinski2014].
+Transfer learning is most useful when pre-training and target datasets are of similar nature [@tag:Yosinski2014].
+For this reason, it is important to search for similar data that is already available and may potentially be used to increase the size of the training set or for pre-training and subsequent fine-tuning on the target data.
```

Comment:
In TL, we also don't always want/need to completely start from scratch and use a similar dataset to pre-train the model ourselves. Many pre-trained models are also off-the-shelf available for further training. Maybe we should add a comment about that? E.g., something like

"As an alternative to pre-training models on larger datasets for transfer learning yourself, you may also be able to obtain pre-trained models from public repositories."

Or is it better not to recommend that (as the publicly available models are mostly "too general")?

Reply:
I had actually assumed that the pre-trained models often would come from a public repository, so if that was not clear then I agree we should make it more explicit.  It may not work all the time, but I see reusing pre-trained public models as a generally-recommended strategy, especially in image analysis (e.g. microscopy) and regulatory genomics.

The kipoi model zoo (https://doi.org/10.1101/375345) is a good source of pre-trained models for regulatory genomics.  For imaging, generic non-bio models are a common starting point.

Reply:
I think both are valid approaches. Also, they could be even combined: (1) obtain model pretrained on e.g., imagenet -> (2) fine tune on large-scale public x-ray dataset -> (3) finetune to your x-ray dataset. I don't have much practical experinence with TL though, and I am not sure what would work better in practice, i.e,. (1) -> (2) -> (3), or (1) -> (3), or (2) -> (3).

 I'd say we should definitely reference the paper and include a link to https://kipoi.org, it looks pretty useful.

Reply:
I just pushed a change that (I hope) clarifies the distinction and applicability of pre trained models and data for pre training.

Reply:
@Benjamin-Lee thanks for addressing this! I think it's a helpful clarification.